### PR TITLE
include: define errnos if not defined for better portablity

### PR DESCRIPTION
**Pull request by wjwithagen**
This requires defining
`EUCLEAN`
`EKEYREJECTED`
which are defined in Linux asm-generic/error.h

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/include/compat.h [hunk 1]
```diff
@@ -79,8 +79,16 @@
 /* get PATH_MAX */
 #include <limits.h>
 
+#ifndef EUCLEAN
+#define EUCLEAN         117	/* Structure needs cleaning */
```
Comment (tchaikov) [inline src/include/compat.h]: could you drop the extra spaces? and drop the comments also. as it's not our responsibility to document the errno. whoever interested in it should consult the manpage, for instance, http://man7.org/linux/man-pages/man3/errno.3.html .

Comment (wjwithagen) [reply src/include/compat.h]: @tchaikov 
Will do, I'll just ad the URl as comment.

Comment (tchaikov) [reply src/include/compat.h]: @wjwithagen i'd suggest not.

Comment (wjwithagen) [reply src/include/compat.h]: @tchaikov 
Perhaps it is a note to myself, as I'm regularly looking for the place where things like these are defined. But I'll remove

File: src/include/compat.h [hunk 2]
```diff
@@ -79,9 +79,16 @@
 /* get PATH_MAX */
 #include <limits.h>
 
+/* Linux error constarts: see http://man7.org/linux/man-pages/man3/errno.3.html */
```
Comment (tchaikov) [inline src/include/compat.h]: please drop this comment. see https://github.com/ceph/ceph/pull/25302#discussion_r

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2018-11-29_01:22:06-rados-wip-kefu2-testing-2018-11-29-0053-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2018-11-29_08:14:41-rados-wip-kefu2-testing-2018-11-29-1356-distro-basic-smithi/

test failures are tracked / caused by

- #19454
- http://tracker.ceph.com/issues/37391

**Comment by wjwithagen** [comment]
@tchaikov 
Thanx,
BTW, I'm of for the weekend, so no more posts until monday earliest.

**Pull request merged**